tests/common/mod.rs

```diff
@@ -254,9 +255,11 @@ pub(crate) enum TestChainSource<'a> {
 pub(crate) enum TestLogWriter {
 	File(FilesystemLoggerConfig),
 	LogFacade(LogFacadeLoggerConfig),
+	Custom(Arc<dyn LogWriter + Send + Sync>),
 }
 
 /// Simple in-memory mock `log` logger for tests.
+#[derive(Debug)]
```

Comment:
`NodeBuilder` is `Debug` because , and it fields an optional `LogWriterConfig`. The config objects has to be `Debug` and custom log writer variant needs to implement it for this reason.